src/test/java/org/terasology/simpleFarming/systems/BushAuthoritySystemTest.java

```diff
@@ -1,57 +1,162 @@
+
+    private Block air;
+    private Block dirt;
+
+    @In
+    EntityManager entityManager;
+    @In
+    WorldProvider worldProvider;
+    @In
+    BlockManager blockManager;
+    @In
+    ModuleTestingHelper helper;
+    @In
+    BlockEntityRegistry blockEntityRegistry;
+
+    @BeforeEach
+    public void initialize() {
+        air = blockManager.getBlock("engine:air");
+        dirt = blockManager.getBlock("CoreAssets:Dirt");
+
+        setBlock(new Vector3i(0, -1, 0), dirt);
+        setBlock(new Vector3i(), air);
+
+        final EntityRef seed = entityManager.create("SimpleFarming:TestSeed");
+        this.plant(seed, new Vector3f());
+
+    }
 
     @Test
-    @Ignore // TODO Seems to enter infinite loop
     public void bushShouldGrowInOrder() {
-        for (int i = 0; i < getFinalGrowthStageIndex(); i++) {
-            assertBushInStage(i);
-            waitForGrowth();
+        EntityRef entity = blockEntityRegistry.getExistingBlockEntityAt(new Vector3i());
+        BushDefinitionComponent component = entity.getComponent(BushDefinitionComponent.class);
+        for (int stage = 0; stage < component.growthStages.size(); stage++) {
+            // verify the the bush is on the current stage
+            assertEquals(stage, component.currentStage);
+            BlockComponent blockComponent = entity.getComponent(BlockComponent.class);
+            String block = BushAuthoritySystem.getGrowthStage(component, stage).getKey();
+            assertEquals(block, worldProvider.getBlock(blockComponent.position).toString());
+
+            final int startStage = component.currentStage;
+            helper.runWhile(() -> component.currentStage == startStage);
         }
     }
 
     @Test
-    @Ignore // TODO Seems to enter infinite loop
     public void harvestingSustainableBushShouldResetGrowthAndDropProduce() {
-        waitUntilHarvestable();
-        assertActionDropsProduce(this::harvestBush);
-        assertBushInStage(getFinalGrowthStageIndex() - 1);
+        EntityRef entity = blockEntityRegistry.getExistingBlockEntityAt(new Vector3i());
+        Assertions.assertNotNull(entity);
+        BushDefinitionComponent component = entity.getComponent(BushDefinitionComponent.class);
+
+        // wait until the bush gets to the final growth state
+        helper.runUntil(() -> (component.currentStage == (component.growthStages.size() - 1)));
+
+        // harvest bush
+        final TestEventReceiver<DropItemEvent> dropSpy = new TestEventReceiver<>(helper.getHostContext(), DropItemEvent.class);
+        entity.send(new ActivateEvent(entity, playerInstigator(), null, null, null, null, 0));
+
+        // check if bush drops produce
+        Assertions.assertTrue(dropSpy.getEntityRefs().stream().anyMatch(k -> k.getParentPrefab().getName().equals(component.produce)));
+
+        // check if bush is at max growth state
+        assertEquals(component.growthStages.size() - 2, component.currentStage);
```

Comment:
So, the max growth stage is one state before the final growth state? This is confusing :roll_eyes: